src/Transport/Sending/Batcher.cs

```diff
@@ -110,5 +110,40 @@ public class BatchedOperation
         public OutgoingMessage Message { get; set; }
 
         public IEnumerable<DeliveryConstraint> DeliveryConstraints { get; set; }
+
+        public long GetEstimatedSize()
+        {
+            const int assumeSize = 256;
+            var standardPropertiesSize = GetStringSizeInBytes(Message.MessageId) +
+                                         assumeSize + // ContentType +
+                                         assumeSize + // CorrelationId +
+                                         4 + // DeliveryCount
+                                         8 + // EnqueuedSequenceNumber
+                                         8 + // EnqueuedTimeUtc
+                                         8 + // ExpiresAtUtc
+                                         1 + // ForcePersistence
+                                         1 + // IsBodyConsumed
+                                         assumeSize + // Label
+                                         8 + // LockedUntilUtc 
+                                         16 + // LockToken 
+                                         assumeSize + // PartitionKey
+                                         8 + // ScheduledEnqueueTimeUtc
+                                         8 + // SequenceNumber
+                                         assumeSize + // SessionId
+                                         4 + // State
+                                         8 + // TimeToLive +
+                                         assumeSize + // To +
+                                         assumeSize;  // ViaPartitionKey;
+
+            var headers = Message.Headers.Sum(property => GetStringSizeInBytes(property.Key) + GetStringSizeInBytes(property.Value));
+            var bodySize = Message.Body.Length;
+            var total = standardPropertiesSize + headers + bodySize;
+
+            const double addTenPercent = 1.1;
```

Comment:
Add this to config, so a user can fine tune it for his scenario. Question: if body type is stream, would this also be necessary?